std/format.d

```diff
@@ -4362,14 +4362,44 @@ T unformatValue(T, Range, Char)(ref Range input, ref FormatSpec!Char spec)
     return parse!T(input);
 }
 
+private T rawRead(T, Range)(ref Range input)
+    if (isSomeString!Range || ElementType!(Range).sizeof == 1)
+{
+    union X
+    {
+        ubyte[T.sizeof] raw;
+        T typed;
+    }
+    X x;
+    foreach (i; 0 .. T.sizeof)
+    {
+        static if (isSomeString!Range)
+        {
+            x.raw[i] = input[0];
```

Comment:
This will not compile if `Range` is a wide string: you'd attempt to assign a wide char (2 or 4 bytes) over a single byte. I suggest you change the constraint for `rawRead` to just `ElementType!(Range).sizeof == 1`. There is a crazier possibility by which you do accept wide strings and you do implement raw reads, but only if `T.sizeof` is a multiple of the character size. Probably we don't need this; most people who'd entertain the idea of a raw read will do so from a stream of bytes or a narrow string.